src/Driver/API/PostgreSQL/ExceptionConverter.php

```diff
@@ -77,6 +78,13 @@ public function convert(Exception $exception, ?Query $query): DriverException
                 return new ConnectionException($exception, $query);
         }
 
+        if (
+            $exception->getSQLState() === 'HY000'
+            && str_contains($exception->getMessage(), 'server closed the connection')
+        ) {
+            return new ConnectionLost($exception, $query);
+        }
```

Comment:
This might only be applicable for recent PGSQL versions...

Reply:
Nah... only PDO has the SQL State set for this error :)

Ooohh I missed working with DBAL :sweat_smile: